src/plugin/WebAppFactoryLuna.h

```diff
@@ -21,11 +21,15 @@
 
 #include <QtCore/QUrl>
 
+#if defined(HAS_AGL_SERVICE)
+class WebAppFactoryLuna : public WebAppFactoryInterface {
```

Comment:
I think it would be just better that you'd add a new WebAppFactoryAGL, that would not depend on any of the plugin boilerplate. In AGL build we'd just use it instead of the Luna factory.

Also, we'd need to have a WebAppWaylandAGL that would take into account the surface ID (so we can go back to not use surfaceId for webOS initialization).